### PR TITLE
Refactor delete_modal_actions template and use it for project column related actions

### LINKED ISSUE #23318: Organization-wide projects bugs
### Description

At the moment, organization-wide projects **are riddled with bugs**, based on preliminary user feedback from 1.19.0-rc1:
They are
- [x] no one (not even the site admin) can edit/delete org projects, see screenshots 1 and 3
- [x] repo projects allow to set the `card preview type` during creation. Organization projects don't, see screenshots 5 and 6
- [x] Org projects ignore the column color, see screenshots 1 and 2, the right column should have the color `#ff0000`.

These issues can be fixed by sharing the code of repo projects with the organization projects.

Furthermore, the following issues are present:
- [x] The organization `Projects` tab misses quite a few tabs as well as the organization settings, see screenshots 1, 3, and 5.
- [ ] When creating an org project, and adding a second column
  - [x] and setting the second one as default, the initially created column disappears permanently, until the second column is deleted again. Happens both for repo- and org-projects. (#23401)
  - [ ] only the second column can be edited, the first one can't, see screenshot 1 and 2. Happens both for repo- and org-projects (#29873)
- [x] If a repo disables projects, its issues cannot be assigned to org-wide projects either. While the current logic for that is clear (`if isPackagesEnabled <list available projects> end`), this would probably still be useful to implement.
- [x] Both organization and repo projects cannot edit the column color anymore
- [x] If you click on the PR that closed an issue inside an org-wide project, you get redirected to `$URL/pull/$ID` instead of `$URL/$USER/$REPO/pull/$ID`

### Screenshot 1: Org project appearance
image
### Screenshot 2: Repo project appearance 
image
### Screenshot 3: org repo projects list
image
### Screenshot 4: Repo projects list
image
### Screenshot 5: New org project
image
### Screenshot 6: New repo project
image

### Gitea Version

1.20.0+dev-90-gea7f0d6fc

### Can you reproduce the bug on the Gitea demo site?

Yes

### Operating System

Linux

### Browser Version

Firefox 110.0

### PR DESCRIPTION
**Pull request by HesterG**
Co-Author: @wxiaoguang 

This PR is to fix https://github.com/go-gitea/gitea/issues/23318#issuecomment-1506275446 . The way to fix this in this PR is to use `delete_modal_actions.tmpl` here both to fix this issue and keep ui consistency (as suggested by [TODO here](https://github.com/go-gitea/gitea/blob/4299c3b7db61f8741eca0ba3d663bb65745a4acc/templates/projects/view.tmpl#L161))

And this PR also refactors `delete_modal_actions.tmpl` and its related styles, and use the template for more modal actions:

1. Added template attributes:
* locale
* ModalButtonStyle: "yes" (default) or "confirm"
* ModalButtonCancelText
* ModalButtonOkText

2. Rename `delete_modal_actions.tmpl` template to `modal_actions_confirm.tmpl` because it is not only used for action modals deletion now.

3. Refactored css related to modals into `web_src/css/modules/modal.css` and improved the styles.

4. Also use the template for PR deletion modal and remove issue dependency modal. 

5. Some modals should also use the template, but not sure how to open them, so mark these modal actions by `{{/* TODO: Convert to base/modal_actions_confirm */}}`

After (Also tested on arc green):

Hovering on the left buttons

Screen Shot 2023-04-23 at 15 17 12

Screen Shot 2023-04-23 at 15 17 21

Test for functionalities:

https://user-images.githubusercontent.com/17645053/233826857-76376fda-022c-42d0-b0f3-339c17ca4e59.mov

### CODE REVIEW

**Comment by wxiaoguang** [comment]
Conflict with #23531 ?

This PR seems not complete, for example:

* The template shouldn't be called `delete_modal_actions.tmpl` anymore, it's not only used for deletion now.
* It's better to make the sub template gets translated texts, instead of passing the keys in
    *  Explicit `{{$.locale.Tr "key"}}` is easier to be grepped and maintained in the futre.
    *  But `{{$.locale.Tr $key}}` is harder to know what key it uses.
* `ExtendedClasses` looks quite hacky and not friendly for maintaining

**Comment by HesterG** [comment]
> Conflict with #23531 ?

I see, then it might not suitable to use template for setting/unsetting default column. So I will only use template for deletion and wait for #23531 to merge first. 

> * The template shouldn't be called `delete_modal_actions.tmpl` anymore, it's not only used for deletion now.

Yes I also noticed that, but i am not sure if the refactor should be done in this PR (Not sure If using the template is a suitable fix for this PR here, If it is a suitable fix, I can add a commit to do the  refactor)

> * It's better to make the sub template gets translated texts, instead of passing the keys in
>   * Explicit `{{$.locale.Tr "key"}}` is easier to be grepped and maintained in the futre.
>   * But `{{$.locale.Tr $key}}` is harder to know what key it uses.

I will change this.

> * `ExtendedClasses` looks quite hacky and not friendly for maintaining

Do you have a suggestion for this?

**Comment by silverwind** [comment]
https://github.com/go-gitea/gitea/pull/23531 was merged.

**Comment by wxiaoguang** [comment]
"page_dimmer_buttons" , I am a little curious about this name, what is "dimmer" and when/why dimmer?

**Comment by HesterG** [comment]
> "page_dimmer_buttons" , I am a little curious about this name, what is "dimmer" and when/why dimmer?

I find it [here](https://fomantic-ui.com/modules/dimmer.html#page-dimmer), which means the buttons are on a page dimmer... And i think this template is not a modal? 

Or maybe `basic-modal-buttons`? 

Just found [modal with class "basic"](https://fomantic-ui.com/modules/modal.html#basic) has this style.

**Comment by silverwind** [comment]
Also, we are inconsistent with horizontal alignment of buttons. Would suggest to always horizontally center them.

Column delete:

image

PR delete:

image

**Comment by HesterG** [comment]
> This does not seem to affect all modals, for example PR delete is still old style. Is it intended that only a subset of modals is altered?

This affects the modal that uses the `basic_modal_actions.tmpl` subtemplate, this one is different because it does not use the template

https://github.com/go-gitea/gitea/blob/7b34cba0b83a3ae7861916054713baf282281fe8/templates/repo/issue/view_content/sidebar.tmpl#L685

Maybe this one should also use the template?

**Comment by silverwind** [comment]
Thanks, but what about button alignment? In your screenshot, they left-align? Can we center them all?

**Comment by HesterG** [comment]
> Thanks, but what about button alignment? In your screenshot, they left-align? Can we center them all?

Yes, I added a selector in https://github.com/go-gitea/gitea/pull/24097/commits/159e22880c2accd452a3198f3db72159ebece0c4 to make the texts left aligned just like in the template.

**Comment by silverwind** [comment]
I do think centered would be best aesthetic-wise, woudln't you agree?

**Comment by HesterG** [comment]
> I do think centered would be best aesthetic-wise, woudln't you agree?

For this delete pull request modal, I agree. But for the modals with shorter contents will be like below. Maybe a little crowded if centered? 

Screen Shot 2023-04-21 at 14 26 16

**Comment by HesterG** [comment]
> use centerd style for buttons

Added a commit to use center style https://github.com/go-gitea/gitea/pull/24097/commits/7545bca0eec5fb96673779058c98e22fa6410e1c

**Comment by silverwind** [comment]
Thanks, I guess one-line text could also be centered, but let's keep such changes for later, don't need to bloat this PR and keep a reduced scope.

**Comment by HesterG** [comment]
This PR is ready for review again.

File: templates/repo/projects/view.tmpl [hunk 1]
```diff
@@ -162,10 +163,12 @@
 									<div class="content center">
 										<label id="default-project-board-content"></label>
 									</div>
-									<div class="text right actions">
-										<button class="ui cancel button">{{$.locale.Tr "settings.cancel"}}</button>
-										<button class="ui primary button default-project-board-button" id="default-project-board-submit"></button>
-									</div>
+									{{$ModalNoText := $.locale.Tr "settings.cancel"}}
+									{{- $ModalYesText := $.locale.Tr "repo.projects.column.set_default" -}}
+									{{- if .Default -}}
```
Comment (wxiaoguang) [inline templates/repo/projects/view.tmpl]: It looks strange, what's the reason behind this `if` ?

Comment (HesterG) [reply templates/repo/projects/view.tmpl]: This is to match with the logic here:

Screen Shot 2023-04-23 at 10 58 59

**Comment by silverwind** [comment]
Let's remove `text-align: left !important;` in `web_src/css/base.css` lint 1157. I think centered text looks much better with the buttons now centered.

Before:
Screenshot 2023-04-23 at 05 31 59

After:
Screenshot 2023-04-23 at 05 32 13

**Comment by wxiaoguang** [comment]
I guess the problem is that Gitea's modal is border-less at the moment, it's not the same as modern OS's dialog.

I will try to do some fine tunes. To make it more likely a modern dialog, and fix some legacy problems.

**Comment by silverwind** [comment]
Yes that would be nice. I personally hate these "full screen" modals, they are way too distracting. A simple box like in your screenshot is more than sufficient.

**Comment by wxiaoguang** [comment]
Actually, Fomantic UI does well in this case. https://fomantic-ui.com/modules/modal.html

Gitea itself made a mess.


image

**Comment by silverwind** [comment]
That said, I think we should completely move all modals into a single shared template with `dict` props, not only the buttons. But I can accept this PR for now, more improvement can come later.

**Comment by wxiaoguang** [comment]
Give me some more time (maybe one day).

**Comment by silverwind** [comment]
Want to do it in this PR?

Another idea would be to use [`<dialog>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog) in place of Fomantic.

**Comment by wxiaoguang** [comment]
> Want to do it in this PR?


Yes, the code `{{- if .Default -}}` looks fragile, I will try.

**Comment by wxiaoguang** [comment]
Almost done, not perfect but much better than before.

## Before

image

image

image


## After

image

image


image

**Comment by HesterG** [comment]
Updated screenshots and video in the description, I think this is ready again.

**Comment by HesterG** [comment]
> That said, I think we should completely move all modals into a single shared template with `dict` props, not only the buttons. But I can accept this PR for now, more improvement can come later.

I think the modal refactor will be in another PR as this PR is already big enough lol.

**Review by wxiaoguang — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
@HesterG @wxiaoguang could you update the issue content so the commit message will look better.

**Comment by wxiaoguang** [comment]
I have no writer permission, so can't edit

**Comment by HesterG** [comment]
Updated the description again to include more information. @wxiaoguang please check if anything needs to be changed.

**Comment by HesterG** [comment]
Updated TODO related description.

**Pull request merged**